compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc

```diff
@@ -6307,7 +6308,36 @@ INTERCEPTOR(int, fclose, __sanitizer_FILE *fp) {
 INTERCEPTOR(void*, dlopen, const char *filename, int flag) {
   void *ctx;
   COMMON_INTERCEPTOR_ENTER_NOIGNORE(ctx, dlopen, filename, flag);
-  if (filename) COMMON_INTERCEPTOR_READ_STRING(ctx, filename, 0);
+
+  if (filename) {
+    COMMON_INTERCEPTOR_READ_STRING(ctx, filename, 0);
+
+#  if !SANITIZER_DYNAMIC
```

Comment:
`if constexpr (!SANITIZER_DYNAMIC)` may work

Reply:
```
error: use of undeclared identifier 'SANITIZER_DYNAMIC'
    if constexpr (!SANITIZER_DYNAMIC) {
                   ^
1 error generated.
```

(It's usually not defined at all.)